### PR TITLE
Add command to serve pach API from a debug dump

### PR DESCRIPTION
**Pull request by PFedak**

### CODE REVIEW

File: src/server/debug/shell/server.go [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+		}
+		return srv.Send(&info)
+	})
+}
+
+func (d *debugDump) listBranch(req *pfs.ListBranchRequest, srv pfs.API_ListBranchServer) error {
+	glob := fmt.Sprintf(commitPatternFormatString, req.Repo.Name)
+	var ci pfs.CommitInfo
+	var found bool
+	branchSet := make(map[string]bool)
```
Comment (acohen4) [inline src/server/debug/shell/server.go:351]: nit: for consistency, should we map to struct{} ?

**Review by acohen4 — APPROVED** [review]
Left some nits/questions but overall LGTM

File: src/internal/ppsutil/util.go [hunk 2]
```diff
@@ -513,3 +517,44 @@ func ListPipelineInfo(ctx context.Context,
...
+			}
+			if !common.MatchDatum(request.DataFilters, msg.Data) {
+				continue
+			}
+		}
+		// Log message passes all filters -- return it
+		msg.Message = strings.TrimSuffix(msg.Message, "\n")
+		if err := send(msg); err != nil {
+			if errors.Is(err, errutil.ErrBreak) {
+				return nil
```
[the remark this answers is not in the document]
Comment (PFedak) [reply src/internal/ppsutil/util.go:554]: Yes, that return should be exiting the loop, but I could switch it to a `break` if that's clearer

File: src/server/debug/shell/server.go [hunk 3]
```diff
@@ -0,0 +1,598 @@
...
+			if errors.Is(err, errutil.ErrBreak) {
+				break
+			}
+			return err
+		}
+	}
+	return nil
+}
+
+func (d *debugDump) globTarProtos(glob string, template proto.Message, onFile func(string) error, onProto func() error) error {
```
[the remark this answers is not in the document]
Comment (PFedak) [reply src/server/debug/shell/server.go]: After realizing this means we'd have to type assert the message everywhere I'm not as eager to do this

**Pull request merged**